lib/tasks/maintenance.rake

```diff
@@ -225,6 +225,40 @@ namespace :maintenance do
     end
   end
 
+  task migration_contract_duration: :environment do
+    puts "Did you set duration boolean on all contract_type ? (y/N)"
+    response = STDIN.gets.chomp.downcase
+    next unless %w[yes y].include?(response)
+    puts "Migrating..."
+
+    # Create missing contract duration
+    (1..11).each do |i|
+      ContractDuration.find_or_create_by!(name: "#{i} mois")
+      ContractDuration.find_or_create_by!(name: "#{i} mois renouvelable")
+    end
+    ContractDuration.find_or_create_by!(name: "1 an")
+    ContractDuration.find_or_create_by!(name: "1 an renouvelable")
+    (2..5).each do |i|
+      ContractDuration.find_or_create_by!(name: "#{i} années")
+      ContractDuration.find_or_create_by!(name: "#{i} années renouvelable")
+    end
+
+    # Assign contract duration to job offer when transition from duration_contract has not been done
+    JobOffer.where.not(duration_contract: nil).map do |job_offer|
+      if job_offer.contract_type.duration
```

Comment:
can you add support of different cases ("an" vs "AN", "ans" vers "ANS") and variant ("années" vs "an")?